Formula/opencascade.rb

```diff
@@ -57,7 +57,7 @@ def install
   end
 
   test do
-    output = shell_output("#{bin}/DRAWEXE -c \"pload ALL\"")
+    output = shell_output("DYLD_LIBRARY_PATH=#{lib} CASROOT=#{prefix}/ #{libexec}/bin/DRAWEXE -c \"pload ALL\"")
```

Comment:
Setting `CASROOT` is acceptable, but not `DYLD_LIBRARY_PATH`: the formula should be compiled in a way that it can find its own libraries. We should fix the root cause

Reply:
It can find its own libraries on Intel. This appears to only be a problem on ARM.

Reply:
I'm not entirely sure why the DYNLD path does not needs to be set on x86, that seems a bit odd. The documentation claims that drawexe should be called through a (broken) helper script that sets dynamic loader's search path on both Linux and macOS. 

What's the preferred solution here? This is effectively just a test/sample application, so fixing the scripts seems like it isn't exactly worth it.

Reply:
> The documentation claims that drawexe should be called through a (broken) helper script

I still think it should be reported as a bug and fixed

Reply:
OK, I think I have figured out what is going on and why it worked on x86. It seems like OpenCASCADE is trying to load the offending libraries dynamically using dlopen rather than linking them directly to the main library. Interestingly, the library relies on the dynamic linker's search path to find the modules. In practice, that means that they install all of the modules (~plugins) in {prefix}/lib even though you aren't supposed to link directly to them. I'm not entirely sure why they are doing that.

The reason this works on x86 is probably that they are sym-linked into /usr/local/lib and end up in the default search path. That's no longer the case on Apple silicon, so the linker won't find them unless we set the rpath appropriately.

It seems like they tried to work around these issues by installing a bunch of wrapper scripts that setup various environment variables (LD_LIBRARY_PATH being is one of them). They seem to rely on quirks like the current directory ('.') being in PATH. IMO, the best way to fix them is to not use them. In fact, the packet installs these helper scripts using generic names (env.sh, draw.sh, custom.sh) in Homebrew's bin directory.

I have fixed this package by setting RPATH at build time for the installed binary.